crypto/ml_dsa/ml_dsa_encoders.c

```diff
@@ -768,7 +766,13 @@ int ossl_ml_dsa_sk_decode(ML_DSA_KEY *key, const uint8_t *in, size_t in_len)
     size_t i, k = params->k, l = params->l;
     PACKET pkt;
 
-    if (key->priv_encoding != NULL || key->pub_encoding != NULL)
+    /* When loading from an explicit key, drop the seed. */
+    OPENSSL_free(key->seed);
+    key->seed = NULL;
+
+    /* Allow the key encoding to be already set to the provided pointer */
+    if ((key->priv_encoding != NULL && key->priv_encoding != in)
```

Comment:
Why does key->priv_encoding == in happen (has it already been processed in this case?)

Reply:
This happens when the key has been decoded, but not yet loaded or imported. A separate pointer could be used, and "moved" (set one clear the other) so that the "prekey" is more explicit. I am not opposed if you think that's warranted, already commented on this...

Reply:
It not particularly easy to follow either way :)

Reply:
If you have a specific plan and time to execute it, I'm not opposed to modest refactoring. What do you have in mind?